src/chttpd/src/chttpd_misc.erl

```diff
@@ -146,14 +150,31 @@ all_dbs_callback({row, Row}, #vacc{resp = Resp0} = Acc) ->
             {ok, Resp1} = chttpd:send_delayed_chunk(Resp0, [Prepend, ?JSON_ENCODE(DbName)]),
             {ok, Acc#vacc{prepend = ",", resp = Resp1}}
     end;
-all_dbs_callback(complete, #vacc{resp = Resp0} = Acc) ->
+all_dbs_info_callback({row, Row}, #vacc{resp = Resp0} = Acc)
+  when Acc#vacc.req#httpd.path_parts =:= [<<"_dbs_info">>] ->
+    Prepend = couch_mrview_http:prepend_val(Acc),
+    DbName = couch_util:get_value(id, Row),
+    case chttpd_util:get_db_info(DbName) of
+        {ok, DbInfo} ->
+            Chunk = [Prepend, ?JSON_ENCODE({[{key,DbName}, {info, {DbInfo}}]})],
+            {ok, Resp1} = chttpd:send_delayed_chunk(Resp0, Chunk),
+            {ok, Acc#vacc{prepend = ",", resp = Resp1}};
+        {error, database_does_not_exist} ->
+            {ok, Acc#vacc{resp = Resp0}};
+        {error, Reason} ->
+            {ok, Resp1} = chttpd:send_delayed_error(Resp0, Reason),
+            {ok, Acc#vacc{resp = Resp1}}
```

Comment:
I wonder if this should be {stop, ...} instead. What do we do in other such examples right after we send an error to the user? Sending the error probably throws an abort so it doesn't even matter.

Reply:
Changed, thanks for the review.